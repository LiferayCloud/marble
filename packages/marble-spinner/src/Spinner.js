import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './Spinner.soy.js';

/**
 * Spinner component.
 */
class Spinner extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
Spinner.STATE = {
  /**
   * Indicates if the spinner rotation is done or not
   * @type {!Boolean}
   */
  isDone: Config.bool(),

  /**
   * The size of the spinner
   * @type {!String|undefined}
   */
  size: Config.oneOf(['small', 'medium', 'large']),

  /**
   * The style of the spinner
   * @type {!String|undefined}
   */
  style: Config.oneOf(['danger', 'success', 'warning']),
};

Soy.register(Spinner, templates);

export {Spinner};
export default Spinner;
