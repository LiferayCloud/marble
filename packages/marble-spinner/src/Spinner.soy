{namespace Spinner}

/**
 *
 */
{template .render}
  {@param? isDone: ?}
  {@param? size: ?}
  {@param? style: ?}

  {let $spinnerClasses kind="text"}
    {if $isDone}{sp}spinner-done{/if}
    {if $size}{sp}spinner-{$size}{/if}
    {if $style}{sp}spinner-{$style}{/if}
  {/let}

  {let $icon kind="text"}
    {if $style == 'danger'}
      icon-12-close-short
    {elseif $style == 'warning'}
      icon-12-exclamation
    {else}
      icon-12-check
    {/if}
  {/let}

  <span class="spinner{$spinnerClasses}">
    {if $isDone}
      <span class="{$icon}"></span>
    {/if}
  </span>
{/template}
