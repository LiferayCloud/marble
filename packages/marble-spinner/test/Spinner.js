import Spinner from '../src/Spinner';

let spinner;

describe('Spinner', () => {
  afterEach(() => {
    if (spinner) {
      spinner.dispose();
    }
  });

  it('should generate the default markup', () => {
    spinner = new Spinner();

    expect(spinner).toMatchSnapshot();
  });

  it('should render with a completion state', () => {
    spinner = new Spinner({
      isDone: true,
    });

    expect(spinner).toMatchSnapshot();
  });

  it('should render with a different size', () => {
    spinner = new Spinner({
      size: 'large',
    });

    expect(spinner).toMatchSnapshot();
  });

  it('should render with warning style', () => {
    spinner = new Spinner({
      style: 'warning',
    });

    expect(spinner).toMatchSnapshot();
  });

  it('should render with danger style', () => {
    spinner = new Spinner({
      style: 'danger',
    });

    expect(spinner).toMatchSnapshot();
  });

  it('should render with success style', () => {
    spinner = new Spinner({
      style: 'success',
    });

    expect(spinner).toMatchSnapshot();
  });

  it('should render with success style and with a completion state', () => {
    spinner = new Spinner({
      isDone: true,
      style: 'success',
    });

    expect(spinner).toMatchSnapshot();
  });
});
