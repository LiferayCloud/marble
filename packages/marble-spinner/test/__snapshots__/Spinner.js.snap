// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should generate the default markup 1`] = `<span class="spinner"></span>`;

exports[`Spinner should render with a completion state 1`] = `
<span class="spinner spinner-done">
  <span class="icon-12-check"></span>
</span>
`;

exports[`Spinner should render with a different size 1`] = `<span class="spinner spinner-large"></span>`;

exports[`Spinner should render with danger style 1`] = `<span class="spinner spinner-danger"></span>`;

exports[`Spinner should render with success style 1`] = `<span class="spinner spinner-success"></span>`;

exports[`Spinner should render with success style and with a completion state 1`] = `
<span class="spinner spinner-done spinner-success">
  <span class="icon-12-check"></span>
</span>
`;

exports[`Spinner should render with warning style 1`] = `<span class="spinner spinner-warning"></span>`;
